sanic/headers.py

```diff
@@ -33,175 +34,310 @@
+    Two separate methods are provided for searching the list:
+    - 'match' for finding the most preferred match (wildcards supported)
+    -  operator 'in' for checking explicit matches (wildcards as literals)
+    """
 
+    def match(self, *mimes: str, accept_wildcards=True) -> Accept:
+        """Find a media type accepted by the client.
+
+        This method can be used to find which of the media types requested by
+        the client is most preferred against the ones given as arguments.
+
+        The ordering of preference is set by:
+        1. The q values on the Accept header, and those being equal,
+        2. The order of the arguments (first is most preferred), and
+        3. The first matching entry on the Accept header.
+
+        Wildcards are matched both ways. A match is usually found, as the
+        Accept headers typically include `*/*`, in particular if the header
+        is missing, is not manually set, or if the client is a browser.
+
+        Note: the returned object behaves as a string of the mime argument
+        that matched, and is empty/falsy if no match was found. The matched
+        header entry `MediaType` or `None` is available as the `m` attribute.
+
+        @param mimes: Any MIME types to search for in order of preference.
+        @param accept_wildcards: Match Accept entries with wildcards in them.
+        @return A match object with the mime string and the MediaType object.
+        """
+        a = sorted(
+            (-acc.q, i, j, mime, acc)  # Sort by -q, i, j
+            for j, acc in enumerate(self)
+            if accept_wildcards or not acc.has_wildcard
+            for i, mime in enumerate(mimes)
+            if acc.match(mime)
+        )
+        return Accept(*(a[0][3:5] if a else ("", None)))
 
-class AcceptContainer(list):
-    def __contains__(self, o: object) -> bool:
-        return any(item.match(o) for item in self)
+    def __str__(self):
+        """Format as Accept header value (parsed, not original)."""
+        return ", ".join(str(m) for m in self)
 
-    def match(
-        self,
-        o: object,
-        *,
-        allow_type_wildcard: bool = True,
-        allow_subtype_wildcard: bool = True,
-    ) -> bool:
-        return any(
-            item.match(
-                o,
-                allow_type_wildcard=allow_type_wildcard,
-                allow_subtype_wildcard=allow_subtype_wildcard,
-            )
-            for item in self
+
+class AcceptContainer(AcceptList):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        deprecation(
+            "The AcceptContainer has been renamed AcceptList. Please update "
+            "your code before v23.9 when AcceptContainer will be removed.",
+            23.9,
         )
 
 
+def _sort_accept_value(accept: MediaType):
+    return (
+        accept.q,
+        len(accept.params),
+        accept.subtype != "*",
+        accept.type_ != "*",
+    )
```

Comment:
Okay, I see in RFC this is technically more correct. Should probably use same rules in AcceptList.match then as well, adding to its sort factors all these. I wonder how much performance effect all this has. If it could go significantly faster by matching q only, I would ignore the criteria for "most specific" as not really needed in real world.

In any case, inverting the values -q, -len(...) etc. is probably helpful to have sorting work without reversal (ascending order for highest precedence first, keeping stable sorting of equally valued originals).

Reply:
Suggest storing sort key in MediaType as property during construction to avoid recalculation, and potentially inverting the values here to avoid reverse=True on sorting phase.

Reply:
I'm fine storing the keys.

Reply:
The sorting of course is "breaking", as previously only q values were used IIRC. I am fine with strict RFC compliance but would like to see the benchmark of how much we are paying for that.

Reply:
This is the exact same function as is in `main`. The only difference is to change the param value from `qvalue` to `q`.